### PR TITLE
Break doi update infinite loop

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: publication-event-dtopublicationdoi-producer/src/test/java/no/unit/nva/doi/event/producer/DynamoDbFanoutPublicationDtoProducerTest.java [hunk 1]
```diff
@@ -133,6 +146,20 @@ void handlerCreatesOutputWithNonEmptyDoiWhenNewImageHasPublicationWithDoi() thro
         assertThat(actualDoi, is(equalTo(doiInResourceFile)));
     }
 
+    @Test
+    void handlerCreatesNoOutputWhenPublicationUpdateDiffersOnlyInModifiedDate() throws JsonProcessingException {
```
Comment (axthosarouris) [inline publication-event-dtopublicationdoi-producer/src/test/java/no/unit/nva/doi/event/producer/DynamoDbFanoutPublicationDtoProducerTest.java:150]: this is the most important test

File: publication-doi-commons/src/main/java/no/unit/nva/publication/doi/update/dto/DoiRegistrarEntryFields.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        doiRegistrarEntryFields.publicationDate = extractPublicationDate(publication);
+        doiRegistrarEntryFields.publisherId = extractPublisherId(publication);
+        doiRegistrarEntryFields.title = extractTitle(publication);
+        doiRegistrarEntryFields.publicationInstance = extractPublicationInstance(publication);
+        doiRegistrarEntryFields.doi = extractPublicationDoi(publication);
+
+        return doiRegistrarEntryFields;
+    }
+
+    @JacocoGenerated
```
Comment (brinxmat) [inline publication-doi-commons/src/main/java/no/unit/nva/publication/doi/update/dto/DoiRegistrarEntryFields.java:45]: Move the overridden methods to the bottom of the class.

File: publication-event-dtopublicationdoi-producer/src/main/java/no/unit/nva/doi/event/producer/DynamoDbFanoutPublicationDtoProducer.java [hunk 3]
```diff
@@ -119,10 +120,19 @@ private boolean shouldPropagateEvent(DynamoEntryUpdateEvent updateEvent) {
...
-
         if (nonNull(newPublication)) {
-            return !newPublication.equals(oldPublication);
+            return newDoiRelatedMetadataDifferFromOld(newPublication, oldPublication);
         }
         return false;
     }
+
+    private boolean newDoiRelatedMetadataDifferFromOld(Publication newPublication, Publication oldPublication) {
+        DoiRegistrarEntryFields doiInfoNew = DoiRegistrarEntryFields.fromPublication(newPublication);
```
Comment (brinxmat) [inline publication-event-dtopublicationdoi-producer/src/main/java/no/unit/nva/doi/event/producer/DynamoDbFanoutPublicationDtoProducer.java:130]: We should talk about this, I think we may be overthinking

File: publication-event-dtopublicationdoi-producer/src/test/java/no/unit/nva/doi/event/producer/DynamoDbFanoutPublicationDtoProducerTest.java [hunk 4]
```diff
@@ -133,6 +146,20 @@ void handlerCreatesOutputWithNonEmptyDoiWhenNewImageHasPublicationWithDoi() thro
...
+        var event =
+            IoUtils.stringFromResources(Path.of(EVENT_PUBLICATION_UPDATED_ONLY_BY_MODIFIED_DATE));
+
+        assertThatEventsDifferOnlyInModifiedDate(event);
```
Comment (brinxmat) [inline publication-event-dtopublicationdoi-producer/src/test/java/no/unit/nva/doi/event/producer/DynamoDbFanoutPublicationDtoProducerTest.java:154]: A service shouldn't modify an internal field directly

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**